From 7bfbaecf819a8b1630dfc8f56126e31f985d5fb3 Mon Sep 17 00:00:00 2001
From: Xavier Claessens <xavier.claessens@collabora.com>
Date: Fri, 29 Jan 2021 19:17:38 -0500
Subject: [PATCH] Windows: Fix symlink privilege error detection

The message is in e.args[1] and not e.args[0] at least with python 3.8.
Should be more future proof like this in case it change again.
---
 conf.d/link_confs.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/conf.d/link_confs.py b/conf.d/link_confs.py
index 0c42efb..5a78d8d 100644
--- a/conf.d/link_confs.py
+++ b/conf.d/link_confs.py
@@ -3,6 +3,7 @@
 import os
 import sys
 import argparse
+import platform
 
 if __name__=='__main__':
     parser = argparse.ArgumentParser()
@@ -26,7 +27,7 @@ if __name__=='__main__':
             break
         except OSError as e:
             # Symlink privileges are not available
-            if len(e.args) == 1 and 'privilege' in e.args[0]:
+            if platform.system().lower() == 'windows' and e.winerror == 1314:
                 break
             raise
         except FileExistsError:
-- 
2.31.1.windows.1

